src/v/storage/tests/readers_cache_test.cc

```diff
@@ -0,0 +1,113 @@
+// Copyright 2022 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "model/fundamental.h"
+#include "seastarx.h"
+#include "storage/readers_cache.h"
+#include "test_utils/fixture.h"
+
+#include <fmt/ostream.h>
+
+namespace storage {
+struct readers_cache_test_fixture {
+    readers_cache_test_fixture()
+      : cache(
+        model::ntp("test", "test", 0), std::chrono::milliseconds(360000)) {}
+
+    bool is_within_locked_range(model::offset begin, model::offset end) {
+        return cache.is_within_locked_range(begin, end);
+    }
+
+    void test_within_range(
+      model::offset::type begin, model::offset::type end, bool in_range) {
+        BOOST_REQUIRE_EQUAL(
+          is_within_locked_range(model::offset(begin), model::offset(end)),
+          in_range);
+    }
+
+    readers_cache cache;
+};
+
+} // namespace storage
+using namespace storage;
+
+FIXTURE_TEST(test_range_is_correctly_locked, readers_cache_test_fixture) {
```

Comment:
Thanks for adding a unit test. Good reminder that we should always write unit tests with new APIs.